tests/php/Forms/TreeMultiselectFieldTest.php

```diff
@@ -4,28 +4,260 @@
 
 use SilverStripe\Assets\File;
 use SilverStripe\Dev\SapphireTest;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\FormTemplateHelper;
 use SilverStripe\Forms\TreeMultiselectField;
 
 class TreeMultiselectFieldTest extends SapphireTest
 {
     protected static $fixture_file = 'TreeDropdownFieldTest.yml';
 
-    public function testReadonly()
+    protected $formId = 'TheFormID';
+    protected $fieldName = 'TestTree';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->form = $this->buildFormMock();
+        $this->field = $this->buildField($this->form);
+        $this->folders = $this->loadFolders();
+
+        $this->folderIds = array_map(
+            static function ($f) {
```

Comment:
I've never seen this before - is the static part needed? Just curious, not a request to change

Reply:
That works in the same way as with static variables.
Since we don't need to ditch and recreate this closure on every setup invocation, we can just make it static and reuse the static instance across all tests.